CP-222 add christmas gift to sponsorship

simgonzalez opened the pull request.

Commit:
CP-206 household.json is missing the mapping for the NaturalFatherAlive

Commit:
CP-206 Correspondance with images needs to call that method with data to avoid a mandatory JSON format

Commit:
CP-206 Correspondance mapping needs to precise a field in the related table for the letters to be created correctly

Commit:
CP-206 FieldOfficeDisasterNotification change on the parameters we use because the data can't be proccessed in the old format

Commit:
CP-206 SubMapping new behaviour

simgonzalez committed:
Merge branch 'CompassionCH:14.0' into 14.0

Commit:
CP-210 Response sent to wordpress API are sent on error but with success code. Modifiy this behaviour to send the error code

simgonzalez committed:
Merge branch 'CompassionCH:14.0' into 14.0

ecino committed:
Update rest_json_api/http.py

Commit:
CP-225 add a comment on the partner when a mandate or a bank account is modified

simgonzalez committed:
Merge branch 'CompassionCH:14.0' into 14.0

Commit:
Merge remote-tracking branch 'origin/14.0' into 14.0

Commit:
CP-206 bis Bytes data type problem when it comes to API request

simgonzalez committed:
Merge branch 'CompassionCH:14.0' into 14.0

Commit:
CP-206 bis don't loose old behaviour on error management

Commit:
Merge remote-tracking branch 'origin/14.0' into 14.0

Commit:
CP-222 Modification on the sponsorship to add a field for the user to add an annual amount of gift for christmas. This works like the birthday gift. A parameter has been added for the due date of the christmas gift

Commit:
CP-222 Add date to update when updating an invoice

Commit:
Merge remote-tracking branch 'origin/14.0' into 14.0

# Conflicts:
#	sponsorship_compassion/models/contract_group.py

Commit:
CP-222 tests for the technical functionnalities on account.move

simgonzalez committed:
Merge branch 'CompassionCH:14.0' into 14.0

ecino commented:
The commit list doesn't look right. Can you do an interactive rebase to fix that?
```bash
git fetch <name_of_compassion_remote>
git rebase -i <name_of_compassion_remote>/14.0
<keep only the relevant commits, delete all other lines>
<save file and continue>
git push -f

Commit:
CP-265
feat(Sponsorship, payment_options): Invoices clean behaviour

When modifying payment method or sponsorship now the invoices will
be updated too with the correct informations. (The payment
reference is still not able to update)

Commit:
Revert "CP-265"

This reverts commit 81f5997581eb863b9c7fdeeb46b141cc6aeb4f92.
